### PR TITLE
Fix: fix equal

### PR DESCRIPTION
**Pull request by Jim8y**
This pr fix the bug that JBoolean(true) != JBoolean(true).

### CODE REVIEW

**Comment by Liaojinghui** [comment]
coverage decreased 51.5%?

File: tests/Neo.Json.UnitTests/UT_JString.cs [hunk 1]
```diff
@@ -48,5 +48,16 @@ public void TestGetEnum()
             woo = s.AsEnum(Woo.Jerry, false);
             Assert.AreEqual(Woo.Jerry, woo);
         }
+
+
```
Comment (shargon) [inline tests/Neo.Json.UnitTests/UT_JString.cs:52]:
```suggestion

```

File: src/Neo.Json/JString.cs [hunk 2]
```diff
@@ -92,5 +92,37 @@ public override JString Clone()
...
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return Value.GetHashCode();
+        }
+
     }
```
Comment (shargon) [inline src/Neo.Json/JString.cs:127]:
```suggestion
    }
```

File: src/Neo.Json/JBoolean.cs [hunk 3]
```diff
@@ -17,6 +17,7 @@ namespace Neo.Json
     /// </summary>
     public class JBoolean : JToken
     {
+
         /// <summary>
```
Comment (shargon) [inline src/Neo.Json/JBoolean.cs:21]:
```suggestion
        /// <summary>
```

File: src/Neo.Json/JNumber.cs [hunk 4]
```diff
@@ -117,5 +117,48 @@ public override JNumber Clone()
...
+
+            var other = obj switch
+            {
+                JNumber jNumber => jNumber,
+                uint u => new JNumber(u),
+                int i => new JNumber(i),
+                ulong ul => new JNumber(ul),
+                long l => new JNumber(l),
+                byte b => new JNumber(b),
+                sbyte sb => new JNumber(sb),
```
Comment (shargon) [inline src/Neo.Json/JNumber.cs:150]:
```suggestion
                byte b => new JNumber(b),
                sbyte sb => new JNumber(sb),
                short s => new JNumber(s),
                ushort us => new JNumber(us),
```

**Comment by shargon** [comment]
> coverage decreased 51.5%?

So random...

**Review by shargon — APPROVED** [review]

**Pull request merged**